src/OVAL/probes/unix/linux/rpmverifyfile_probe.c

```diff
@@ -229,51 +244,75 @@ static int rpmverify_collect(probe_ctx *ctx,
 			fi = rpmfiNew(g_rpm->rpmts, pkgh, tag[i], 1);
 
 		  while (rpmfiNext(fi) != -1) {
-				res.file = oscap_strdup(rpmfiFN(fi));
+				current_file = rpmfiFN(fi);
+				current_file_realpath = oscap_realpath(current_file, NULL);
 		    res.fflags = rpmfiFFlags(fi);
 		    res.oflags = omit;
 
 		    if (((res.fflags & RPMFILE_CONFIG) && (flags & RPMVERIFY_SKIP_CONFIG)) ||
 					((res.fflags & RPMFILE_GHOST)  && (flags & RPMVERIFY_SKIP_GHOST))) {
-					free(res.file);
+					free(current_file_realpath);
 					continue;
 				}
 
 		    switch(file_op) {
 		    case OVAL_OPERATION_EQUALS:
-					if (strcmp(res.file, file) != 0) {
-						free(res.file);
+					if (strcmp(current_file, file) != 0 &&
+							current_file_realpath && file_realpath &&
+							strcmp(current_file_realpath, file_realpath) != 0) {
+						free(current_file_realpath);
 						continue;
 					}
+					res.file = oscap_strdup(file);
 		      break;
 		    case OVAL_OPERATION_NOT_EQUAL:
-					if (strcmp(res.file, file) == 0) {
-						free(res.file);
+					if (strcmp(current_file, file) == 0 ||
+							(current_file_realpath && file_realpath &&
+							strcmp(current_file_realpath, file_realpath) == 0)) {
+						free(current_file_realpath);
 						continue;
 					}
+					res.file = current_file_realpath ? current_file_realpath : strdup(current_file);
 		      break;
 		    case OVAL_OPERATION_PATTERN_MATCH:
-		      ret = pcre_exec(re, NULL, res.file, strlen(res.file), 0, 0, NULL, 0);
-
-		      switch(ret) {
-		      case 0: /* match */
-			break;
-		      case -1:
-			/* mismatch */
-			free(res.file);
-			continue;
-		      default:
-			dE("pcre_exec() failed!");
-			ret = -1;
-			free(res.file);
-			goto ret;
-		      }
-		      break;
+					match1 = pcre_exec(re, NULL, current_file, strlen(current_file), 0, 0, NULL, 0);
+					if (match1 < -1) {
+						ret = -1;
+						dE("pcre_exec() failed!");
+						free(current_file_realpath);
+						goto ret;
+					}
+					if (current_file_realpath) {
+						match2 = pcre_exec(re, NULL, current_file_realpath, strlen(current_file_realpath), 0, 0, NULL, 0);
```

Comment:
Maybe we don't have to change the `case OVAL_OPERATION_PATTERN_MATCH` at this moment because we don't need it now and we haven't created any OVAL content on which we would test it.